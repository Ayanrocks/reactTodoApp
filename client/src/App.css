* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

a,
a:link,
a:visited {
  color: #fff;
  text-decoration: none;
}

html,
body {
  height: 100vh;
}
