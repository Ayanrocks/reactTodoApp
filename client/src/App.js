import React, { Component } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Home from "./components/Home";
import Navbar from "./components/Navbar";
import todoEdit from "./components/todoEdit";

class App extends Component {
  render() {
    return (
      <Router>
        <Navbar />
        <Switch>
          <Route path="/dashboard" component={Home} />
          <Route path="/todo/:id/edit" component={todoEdit} />
        </Switch>
      </Router>
    );
  }
}

export default App;
