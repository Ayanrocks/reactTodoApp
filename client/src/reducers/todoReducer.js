import { FETCH_TODOS, FETCH_BUCKETS } from "../actions/types";

const initialState = {
  todos: [],
  buckets: []
};

export default (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TODOS:
      return { ...state, todos: action.payload };
    case FETCH_BUCKETS:
      return { ...state, buckets: action.payload };
    default:
      return state;
  }
};
