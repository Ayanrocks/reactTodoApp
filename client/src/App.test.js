import React from "react";
import App from "./App";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";
import renderer from "react-test-renderer";

it("App renders correctly", () => {
  const mockStore = createStore(reducers, {}, applyMiddleware(reduxThunk));
  const tree = renderer
    .create(
      <Provider store={mockStore}>
        <App />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
