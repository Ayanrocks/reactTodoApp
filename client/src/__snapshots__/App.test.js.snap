// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <div>
    <div
      className="container mt-5"
    >
      <div
        className="row"
      />
    </div>
    <nav
      className="fab-container undefined"
    >
      <button
        className="fab-item undefined "
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f44336",
            "border": "none",
            "color": "#fafafa",
            "textDecoration": "none",
          }
        }
        tooltip="Create Todo"
      >
        <i
          className="fa fa-sticky-note"
        />
      </button>
      <button
        className="fab-item undefined "
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#f44336",
            "border": "none",
            "color": "#fafafa",
            "textDecoration": "none",
          }
        }
        tooltip="Create Bucket"
      >
        <i
          className="fa fa-beer"
        />
      </button>
      <button
        className="fab-item undefined fab-rotate"
        style={
          Object {
            "backgroundColor": "#f44336",
            "border": "none",
            "color": "#fafafa",
            "textDecoration": "none",
          }
        }
        tooltip="Add"
      >
        <i
          className="fa fa-plus"
        />
      </button>
    </nav>
  </div>,
  <nav
    className="navbar fixed-top navbar-expand-md navbar-light bg-danger"
  >
    <a
      className="navbar-brand text-light"
      href="/"
      onClick={[Function]}
    >
      React Todo App
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
  </nav>,
]
`;
